client/src/components/GameComponent.tsx

```diff
@@ -16,23 +17,24 @@ export const GameConfig = {
 interface GameComponentProps {
   connection: HathoraConnection | undefined;
   token: Token;
-  sessionMetadata: {
-    serverUrl?: string;
-    winningScore: number;
-    isGameEnd?: boolean;
-  };
+  sessionMetadata: SessionMetadata | undefined;
+  isNicknameAcked: boolean;
 }
 export function GameComponent(props: GameComponentProps) {
-  const { connection, token, sessionMetadata } = props;
-  if ((token != null && connection != null) || sessionMetadata.isGameEnd) {
+  const { connection, token, sessionMetadata, isNicknameAcked } = props;
+  const [sceneRendered, setSceneRendered] = React.useState<boolean>(false);
+  if (
```

Comment:
One small issue I see is that when I refresh the page, I have to enter a nickname again, I assume `isNicknameAcked` becomes false? I think would be nice to cache that so if I refresh I still stay in as the same player (keep my score). Maybe that's scope creep though, because I don't think it worked that way prior anyways.

Reply:
yeah so my thinking with that is i want you to be able to change your nickname without going into sessionStorage, so I pre-fill your name and you can just hit enter if you want to keep the same one

is it not pre-filling your name?

Reply:
No it is. But as a player I am sad that I lose my score progress. But I assume that isn't a trivial change, so prob not worth it for this demo game